cmd/nvidia-ctk/cdi/generate/generate.go

```diff
@@ -269,6 +274,8 @@ func (m command) generateSpec(opts *options) (spec.Interface, error) {
 		nvcdi.WithLibrarySearchPaths(opts.librarySearchPaths.Value()),
 		nvcdi.WithCSVFiles(opts.csv.files.Value()),
 		nvcdi.WithCSVIgnorePatterns(opts.csv.ignorePatterns.Value()),
+		// We set the following to allow for dependency injection:
+		nvcdi.WithNvmlLib(opts.nvmllib),
```

Comment:
Is this going to cause problems? I have a vague memory of us purposefully *not* relying on NVML in this subcommand. I also don't quite see why it's necessary yet. Maybe that will become clearer further down.

Reply:
It never became clear. What am I missing?